### PR TITLE
Cleanup MIPS purecap exception handlers

### PR DESCRIPTION
**Pull request by qwattash**
This set of changes will reduce the diff to introduce in dev.

### CODE REVIEW

File: sys/mips/include/cheriasm.h [hunk 1]
```diff
@@ -576,12 +559,33 @@
...
+	.set push;							\
+	.set noat;							\
+	lui	tmp, %pcrel_hi(_CHERI_CAPABILITY_TABLE_-8);		\
+	daddiu	tmp, tmp, %pcrel_lo(_CHERI_CAPABILITY_TABLE_-4);	\
+	cgetpccincoffset dst, tmp;					\
+	.set pop
+
+#define	GET_ABS_CAPTABLE_PTR(dst, tmp)					\
+	ABSRELOC_LA(tmp, _CHERI_CAPABILITY_TABLE_);			\
+	cgetpccincoffset dst, tmp
```
Comment (arichardson) [inline sys/mips/include/cheriasm.h:399]: Should this be setaddr?

Comment (arichardson) [reply sys/mips/include/cheriasm.h:399]: Also is this even needed? Can't we use the pcrel version in `MipsCache`?

**Review by arichardson — APPROVED** [review]
Haven't reviewed in massive detail, but looks reasonable to me.

File: sys/mips/include/cheriasm.h (hunk 1 above)
Comment (qwattash) [reply sys/mips/include/cheriasm.h:399]: Should definitely be setaddr. We need it for all uses in `VECTOR(xxx)` as that code is moved at a different address. `MipsCache` is one of those.

Comment (arichardson) [reply sys/mips/include/cheriasm.h]: Thanks, that makes sense.

**Pull request merged**